app/src/main/res/layout/home_activity.xml

```diff
@@ -1,26 +1,51 @@
 <?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:focusable="true"
+    android:focusableInTouchMode="true"
+    android:orientation="vertical">
 
-    <android.support.v7.widget.RecyclerView
-        android:id="@+id/recyclerView"
+    <EditText
+        android:id="@+id/etSearch"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:clipToPadding="false"
-        android:paddingBottom="12dp"
-        android:paddingTop="12dp"
-        app:layoutManager="android.support.v7.widget.LinearLayoutManager"
-        tools:itemCount="10"
-        tools:listitem="@layout/movie_item" />
+        android:layout_height="50dp"
+        android:hint="@string/search"
+        android:textSize="@dimen/dimenMediumText"
+        android:inputType="text"/>
 
-    <ProgressBar
-        android:id="@+id/progressBar"
-        android:layout_width="wrap_content"
+    <TextView
+        android:id="@+id/tvLog"
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:layout_centerInParent="true"
-        tools:visibility="gone" />
+        android:gravity="center"
+        android:textColor="@color/colorWarning"
+        android:textSize="@dimen/dimenMediumText"
+        android:textStyle="bold"
+        android:visibility="gone"/>
+
+    <RelativeLayout
```

Comment:
What do you think of `RelativeLayout`? Considering the (very) variable sizes of screens on Android phones, what is your suggested layout (Relative, Linear, Frame...) to use?

Reply:
I like to use RelativeLayouts as it positions the elements relativelly to its parent or to another elements, but it`s necessary to prevent the layout against the variable sizes of screens, so it`s better to use ConstraintLayout. This layout allows you to specify the position and size for each view according to spacial relationships with other views in the layout.